packages/api-client/src/store/requests.ts

```diff
@@ -118,9 +121,108 @@ export function extendedRequestDataFactory(
     requestMutators.delete(request.uid)
   }
 
+  /** Duplicate Request */
+  const duplicateRequest = (requestData: Request, collectionUid: Collection['uid']) => {
+    /**
+     * Zod already creates a deep clone during safe parse,
+     * so we can avoid making a structured clone.
+     */
+    const newRequest = schemaModel(requestData, requestSchema, false)
+    if (!newRequest) return console.error('[REQUEST MUTATOR]: Invalid request data.', newRequest)
+
+    /**
+     * Assign a new UID to the request.
+     */
+    newRequest.uid = nanoid()
+
+    /**
+     * Make it easier to see that it's a duplicate request.
+     */
+    if (newRequest.summary) newRequest.summary += ' #1'
```

Comment:
what do you think about sticking to append `Copy` here insted of `#1` since its not incrementing anyway?

Reply:
ideally, add ` (Copy)`, but also increment, so:

* ` (Copy)`, and if that exists:
* ` (Copy 2)`, and if that exists:
* ` (Copy 3)` …